src/Chapter2.hs

```diff
@@ -606,7 +607,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate (x:xs) = replicate 2 x ++ duplicate xs
```

Comment:
It is more efficient to construct lists using the cons operator directly without introducing intermediate values. 🚀 
```suggestion
duplicate (x:xs) = x : x : duplicate xs
```